pages/monsters/[id].vue

```diff
@@ -1,12 +1,36 @@
 <template>
-  <main v-if="monster" class="docs-container container">
+  <main v-if="monster" class="docs-container container" :data-mode="mode">
     <!-- TITLE -->
-    <h1>{{ monster.name }}</h1>
-    <img v-if="monster.img_main" :src="monster.img_main" class="img-main" />
+    <div class="flex items-end justify-between gap-8">
+      <h1 class="flex-auto">{{ monster.name }}</h1>
+
+      <div class="flex flex-none items-start">
+        <button
+          class="flex items-center gap-1 rounded-md p-1 text-xs text-blood outline outline-1 outline-blood hover:bg-blood hover:text-white"
+          @click="toggleMode()"
+        >
+          <Icon
+            :name="
+              mode === 'compact'
+                ? 'heroicons:arrows-pointing-out'
+                : 'heroicons:arrows-pointing-in'
+            "
+          />
+          {{ mode === 'compact' ? 'Regular statblock' : 'Compact statblock' }}
+        </button>
+      </div>
+    </div>
+    <img
+      v-if="mode !== 'compact' && monster.img_main"
+      :src="monster.img_main"
+      class="img-main"
+    />
     <p class="italic">
       <span>
         {{
-          `${monster.size} ${monster.type} ${monster.subtype}, ${monster.alignment}`
+          `${monster.size} ${monster.type}${
+            monster.subtype ? ` (${monster.subtype})` : ''
+          }, ${monster.alignment}`
```

Comment:
Took the liberty of putting the subtype in brackets to follow the WOTC treatment